### PR TITLE
Remove `log_text`, and `log_image` from `LightningLoggerBase` API 

### LINKED ISSUE #11837: [RFC] Remove `log_text` and `log_image` from `LightningLoggerBase` API
## Proposed refactor

Remove these methods from the `LightningLoggerBase` API:
https://github.com/PyTorchLightning/pytorch-lightning/blob/8d23f6287adf1e1040948802e6649c4ae2b3eda1/pytorch_lightning/loggers/base.py#L193-L205

### Motivation
`log_text` and `log_image` are both only overridden on 2 loggers: `WandbLogger` and `NeptuneLogger`.

For the rest of the loggers, since these methods are on the Base API it seems to imply that you can call them, but if you do for most loggers you will get a `NotImplementedError`. This is not a good user experience and it would be better to just keep these methods specific to `WandbLogger` and `NeptuneLogger` so it is clear that they can only be called for those loggers.

Similar motivation to https://github.com/PyTorchLightning/pytorch-lightning/issues/11234

From https://github.com/PyTorchLightning/pytorch-lightning/issues/11234: "A good rule of thumb is that the only required interfaces for the Logger are what the Trainer directly calls. Everything else can be pushed to various implementations". Since the trainer does not call `log_text` or `log_image` these do not need to be required on the Logger API.

### Additional Context
These were added in https://github.com/PyTorchLightning/pytorch-lightning/pull/9545

cc @justusschock @awaelchli @rohitgr7 @tchaton @borda @edward-io @ananthsub @kamil-kaczmarek @Raalsky @Blaizzy @akihironitta

### PR DESCRIPTION
**Pull request by akashkw**
## What does this PR do?
`log_text`, and `log_image`, are not overridden in many logger implementations. They have now been removed.

Fixes #11837

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -1240,7 +1240,8 @@ def _log_hyperparams(self) -> None:
 
             if hparams_initial is not None:
                 self.logger.log_hyperparams(hparams_initial)
-            self.logger.log_graph(self.lightning_module)
+            if hasattr(self.logger, "log_graph"):
+                self.logger.log_graph(self.lightning_module)
```
Comment (akashkw) [inline pytorch_lightning/trainer/trainer.py:1244]: Would it be better to have a try-except instead of using `hasattr`?

**Review by edward-io — APPROVED** [review]
Looks good!

**Review by daniellepintz — APPROVED** [review]
Nice!

**Review by ananthsub — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Comment by daniellepintz** [comment]
Looks ready to be merged!

**Pull request merged**